Improve inserter style

jasmussen opened the pull request:
## Description

Aims to fix #1497.

GIF:

![inserter](https://user-images.githubusercontent.com/1204802/31817423-2a9d508c-b594-11e7-83cb-c6b7d08a37a1.gif)

This moves the tabs to the top, it changes the style of the inserter a bit, and improves the focus styles per #3039. 

Note: I can't tab into the Tabs — pun not intended. Not sure why this is, would appreciate advice.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

Commit:
Polish inserter style, move tabs up top

Commit:
Make heading separators sticky positioned.

Commit:
Revert the sticky header per past discussions.

Commit:
Fix issue with search box border.

afercia commented:
@jasmussen nice. Since the tabs disappear when searching, navigation from the search field to the results doesn't have any "barrier" and this seems a good thing to me. 
If this goes in, please do feel free to close https://github.com/WordPress/gutenberg/issues/1588

No need I review it, I'd just like to see the Inserter in a stable state so we can address accessibility 🙂  I've played a bit with `NavigableMenu` and I think with a few adjustments it could be easily reused for the Inserter and allow to remove a lot of code in `InserterMenu`. There's https://github.com/WordPress/gutenberg/issues/1823 for this. Please go ahead! :shipit:

mtias commented on components/popover/style.scss:
```diff
@@ -70,7 +70,7 @@
 
 	@include break-medium {
-		width: 300px;
+		width: 320px;
```
Is this needed for all popovers or should apply only to the inserter?

jasmussen replied:
Great point. I moved the width stuff around to better spots. Had to also give a width to the publish dropdown now as it inherited teh width from here. 

![screen shot 2017-10-21 at 09 43 33](https://user-images.githubusercontent.com/1204802/31849546-7c4f8796-b644-11e7-913a-83f90947a497.png)

Commit:
Move inserter width to a better spot.

This also adds a width to the publish dropdown, which relied on the width given by the inserter.

jasmussen commented:
Given then thumbs up on this PR, and the most recent width CSS fix, I think this branch is good to merge!

However since today — since _right now_ — is release day for 1.5, I'm not going to just merge haphazardly. @pento if you'd like this PR in, it's got my stamp of approval as working, but up to you whether you want to include in this release. Feel free to merge if yes, otherwise I'll do it next week and have it in 1.6.

pento commented:
Let's put it in 1.6, just in case any regressions crop up.

jasmussen commented:
👍👍

jasmussen commented:
1.5 is released, which means this one can go in!
